### PR TITLE
fix(sql): allow using more data types in PARTITION BY of LATEST ON queries

### PR DESCRIPTION
**Pull request by jerrinot**
PR allows using columns of following data types in PARTITON BY clause of LATEST ON : 
- BYTE 
- DATE 
- TIMESTAMP
- FLOAT
- DOUBLE
- GEOHASH

This was an omission. It's allowed with LONGs so I see no good reason why it should not be allowed with TIMESTAMPs.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|

**Review by ideoma — DISMISSED** [review]

File: core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java [hunk 1]
```diff
@@ -4424,6 +4424,7 @@ private int prepareLatestByColumnIndexes(ObjList<ExpressionNode> latestBy, Recor
                     case ColumnType.SHORT:
                     case ColumnType.INT:
                     case ColumnType.LONG:
+                    case ColumnType.TIMESTAMP:
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:4427]: nit: if we support TIMESTAMP, we should do the same for DATE as both are `long` internally.

Comment (jerrinot) [reply core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:4427]: I can do that, yes.

Comment (bziobrowski) [reply core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:4429]: I extended the change to BYTE, DATE, FLOAT, DOUBLE & GEOHASH .

Comment (jerrinot) [reply core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:4429]: Resolved thanks to @bziobrowski

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|

**Pull request merged**